### PR TITLE
Move preload script to a dedicated class

### PR DESCRIPTION
**Pull request by sadym-chromium**
Preparation for channels in preload scripts. Part 2.
* Move preload script logic to a dedicated class.
* Fix a couple of tests.
* Removed support of sandboxes, as [Page.createIsolatedWorld](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-createIsolatedWorld) is needed for that. And after such a call it's too late.
* Don't wait for the initial run of the preload script to be finished.

### CODE REVIEW

File: tests/conftest.py
Comment (thiagowfx) [inline tests/conftest.py:126]: Use a docstring inside the `def` method. This integrates with `pytest --fixtures` and with python docs that are surfaced in IDEs.

File: tests/conftest.py
Comment (thiagowfx) [inline tests/conftest.py:131]: You can call the inner function `read_sorted_messages` as well (I've seen this pattern in WPT fixtures IIRC)

**Comment by Lightning00Blade** [comment]
Can we maybe hold off merging this, as I want to make release after #850 is merged and a release is made. As I am not sure how it will affect Puppeteer.

File: src/bidiMapper/domains/context/PreloadScriptStorage.spec.ts
Comment (thiagowfx) [inline src/bidiMapper/domains/context/PreloadScriptStorage.spec.ts:111]: Could you keep the same pattern as it used to be previously, by comparing the whole array, instead of only `[0]`?

File: src/bidiMapper/domains/context/bidiPreloadScript.ts
Comment (thiagowfx) [inline src/bidiMapper/domains/context/bidiPreloadScript.ts:106]: Add:

`XXX: Make it possible to evaluate in a specific sandbox / worldName` (or similar)

File: src/bidiMapper/domains/context/bidiPreloadScript.ts
Comment (thiagowfx) [inline src/bidiMapper/domains/context/bidiPreloadScript.ts:49]: Document this

File: src/bidiMapper/domains/context/bidiPreloadScript.ts
Comment (thiagowfx) [inline src/bidiMapper/domains/context/bidiPreloadScript.ts:81]: I'd argue this can be less strict than `Set`.

`Iterable` is more forgiving. Or just a plain array.

It's totally fine to create a `Set` within this function for the sake of deduplication, but let's make the interface more forgiving

(i.e. basically https://en.wikipedia.org/wiki/Robustness_principle)

**Review by thiagowfx — CHANGES_REQUESTED** [review]

File: src/bidiMapper/domains/context/bidiPreloadScript.ts
Comment (thiagowfx) [reply src/bidiMapper/domains/context/bidiPreloadScript.ts:110]: True, it's redundant, I think it would just be clearer in terms of TODO for us, to remember what specific part of CDP we want to extend.

I am fine if you prefer to omit it.

File: src/bidiMapper/domains/context/bidiPreloadScript.ts
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/bidiPreloadScript.ts:81]: done

File: src/bidiMapper/domains/context/bidiPreloadScript.ts
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/bidiPreloadScript.ts:49]: done

File: src/bidiMapper/domains/context/PreloadScriptStorage.spec.ts
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/PreloadScriptStorage.spec.ts]: done

File: src/bidiMapper/domains/context/browsingContextProcessor.ts
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextProcessor.ts]: done

File: tests/conftest.py
Comment (sadym-chromium) [reply tests/conftest.py]: done

File: tests/conftest.py
Comment (sadym-chromium) [reply tests/conftest.py]: done

**Review by thiagowfx — APPROVED** [review]

**Pull request merged**
